packages/workflow-compile/index.js

```diff
@@ -26,12 +26,17 @@ async function compile(config) {
       const Compile = SUPPORTED_COMPILERS[name];
       if (!Compile) throw new Error("Unsupported compiler: " + name);
 
-      const compileMethod =
-        config.all === true || config.compileAll === true
-          ? Compile.all
-          : Compile.necessary;
-
-      return await compileMethod(config);
+      if (config.all === true || config.compileAll === true) {
+        return await Compile.all(config);
+      } else if (config.paths && config.paths.length > 0) {
```

Comment:
There's an opportunity for a stricter test. 
`Array.isArray(config.paths) && config.paths.length > 0)` would fail if `configs.paths` is not an Array, whereas the current test would pass if `config.paths` is truthy and doesn't guarantee `paths` has a `length` property.

Reply:
Do you think this is worth doing? Do you think it will provide value? You are really just providing a stricter validation step?

Reply:
It's your call, I just wanted to bring it to your attention. I think it is more correct to check for an Array type because a truthy value doesn't imply Array. OTOH, since we control how `config.paths` is populated, does it really matter?